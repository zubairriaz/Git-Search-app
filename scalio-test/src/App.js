import React from "react";
import Container from "./container/container";
import { StoreProvider } from "./store/Store";
import { initialState, appReducer } from "./store/reducer";

function App() {
	return (
		<React.Fragment>
			<StoreProvider initialState={initialState} reducer={appReducer}>
				<Container></Container>
			</StoreProvider>
		</React.Fragment>
	);
}

export default App;
