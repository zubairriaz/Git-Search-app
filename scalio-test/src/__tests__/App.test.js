import React from 'react';

import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';

import App from '../App';

test('Check If App Loads succesfully', () => {
  const app = shallow(<App />);
  expect(app).toMatchSnapshot();
});

test('Check If It has Submit Button', () => {
  render(<App />);
  const linkElement = screen.getByText(/Submit/i);
  expect(linkElement).toBeInTheDocument();
});

test('Check If It has input Element', () => {
  render(<App />);
  const input = screen.queryByPlaceholderText(/Enter Login Text/i);
  expect(input).toBeInTheDocument();
});


