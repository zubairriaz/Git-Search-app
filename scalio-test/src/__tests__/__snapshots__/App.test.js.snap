// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check If App Loads succesfully 1`] = `ShallowWrapper {}`;
